static/app/components/onboarding/docWithProductSelection.tsx

```diff
@@ -8,31 +8,95 @@ import LoadingError from 'sentry/components/loadingError';
 import LoadingIndicator from 'sentry/components/loadingIndicator';
 import {DocumentationWrapper} from 'sentry/components/onboarding/documentationWrapper';
 import {MissingExampleWarning} from 'sentry/components/onboarding/missingExampleWarning';
-import {PRODUCT, ProductSelection} from 'sentry/components/onboarding/productSelection';
+import {
+  DisabledProduct,
+  PRODUCT,
+  ProductSelection,
+} from 'sentry/components/onboarding/productSelection';
 import {PlatformKey} from 'sentry/data/platformCategories';
 import platforms from 'sentry/data/platforms';
 import {t} from 'sentry/locale';
-import {Organization, Project} from 'sentry/types';
+import {Project} from 'sentry/types';
 import {OnboardingPlatformDoc} from 'sentry/types/onboarding';
 import {trackAnalytics} from 'sentry/utils/analytics';
 import getDynamicText from 'sentry/utils/getDynamicText';
 import {useApiQuery} from 'sentry/utils/queryClient';
+import useOrganization from 'sentry/utils/useOrganization';
 import SetupIntroduction from 'sentry/views/onboarding/components/setupIntroduction';
 import {SetupDocsLoader} from 'sentry/views/onboarding/setupDocsLoader';
 
+function getDefaultSelectedProducts({
+  hasSessionReplay,
+  hasPerformance,
+}: {
+  hasPerformance: boolean;
+  hasSessionReplay: boolean;
+}) {
+  const defaultSelectedProducts: PRODUCT[] = [];
+
+  if (hasSessionReplay) {
+    defaultSelectedProducts.push(PRODUCT.SESSION_REPLAY);
+  }
+
+  if (hasPerformance) {
+    defaultSelectedProducts.push(PRODUCT.PERFORMANCE_MONITORING);
+  }
+
+  return defaultSelectedProducts;
+}
+
+function getDisabledProducts({
+  hasSessionReplay,
+  hasPerformance,
+  hasRightsToUgradePlan,
+}: {
+  hasPerformance: boolean;
+  hasRightsToUgradePlan: boolean;
+  hasSessionReplay: boolean;
+}) {
+  const disabledProducts: DisabledProduct[] = [];
+
+  if (!hasSessionReplay) {
+    const reason = hasRightsToUgradePlan
+      ? t('To use Session Replay, update your organizations plan to its latest version.')
+      : t(
+          'To use Session Replay, request an owner in your organization to update its plan to the latest version.'
+        );
```

Comment:
I think that `hasRightsToUgradePlan` and messaging around upgrading plans do not belong to the sentry repo.
It should live in getsentry as OSS does not have the concept of plans.

Reply:
you are right...we will have to move this logic to `getsentry`. Thanks for the feedback... will update the code